src/lib/core/a11y/focus-key-manager.ts

```diff
@@ -7,21 +7,33 @@
  */
 
 import {QueryList} from '@angular/core';
+import {INPUT_KEY_RANGE_START, INPUT_KEY_RANGE_END} from '../core';
 import {ListKeyManager, CanDisable} from './list-key-manager';
 
 /**
  * This is the interface for focusable items (used by the FocusKeyManager).
- * Each item must know how to focus itself and whether or not it is currently disabled.
+ * Each item must know how to focus itself, whether or not it is currently disabled
+ * and be able to supply it's label.
  */
 export interface Focusable extends CanDisable {
   focus(): void;
+  getFocusableLabel?(): string;
 }
 
+/**
+ * Time in ms to wait after the last keypress to focus the active item.
+ */
+export const FOCUS_KEY_MANAGER_DEBOUNCE_INTERVAL = 200;
+
 
 export class FocusKeyManager extends ListKeyManager<Focusable> {
+  private _timer: number | null;
+  private _pressedInputKeys: number[] = [];
+  private _hasLabelFn: boolean;
 
   constructor(items: QueryList<Focusable>) {
     super(items);
+    this._hasLabelFn = items && items.first && 'getFocusableLabel' in items.first;
```

Comment:
It's been a while since I did this. Revisiting it, it makes sense to require the `getFocusableLabel` as a part of the `Focusable` interface.